### PR TITLE
fix(E1): profile save/clone pipeline — #198, #209, #223, #234

### PR DESCRIPTION
**Pull request by mnbf9rca**
## E1 · Profile save/clone pipeline (Wave 1, epic #263)

Fixes #198
Fixes #209
Fixes #223
Fixes #234

All four defects were re-verified against current `main` by independent adversarial verification agents before fixing (every handover line citation still held). Fixes follow TDD: each regression test was written first and observed failing for the expected reason.

### #198 (critical) — stale snapshot breaks background scheduled blocking
`TriggerConfigurationModel.saveToProfile` now refreshes the app-group `ProfileSnapshot` after writing trigger/schedule fields onto the model. Previously `createProfile`/`updateProfile` snapshotted **before** `finalizeSave` applied trigger edits, so the `FoqosDeviceMonitor` extension read a stale snapshot at interval start (`ScheduleTimerActivity` logs "no schedule found") and scheduled blocking silently never started until the app was foregrounded. `saveToProfile`'s only call site is `finalizeSave`, so the placement is behavior-identical to snapshotting in the view but unit-testable.

### #209 (high) — cloned profile never gets a snapshot or sync push
`cloneProfile` now writes the clone's snapshot (after trigger copy and migration, after a successful `context.save()`), matching `createProfile`'s "so extensions can read it immediately" behavior. The Duplicate flow additionally pushes the clone via `SyncCoordinator.shared.pushProfile`, mirroring the existing save and delete flows, so the clone reaches CloudKit before its first manual edit.

### #223 (medium) — cloning an unmigrated V1 profile produced a permanently broken clone
`cloneProfile` now copies `source.profileSchemaVersion` and runs `migrateToV2IfNeeded()`, so a V1 source's `blockingStrategyId` is translated into startable triggers on the clone immediately. For V2 sources this is a no-op; V3+ sources now keep their version on the clone (previously stamped V2 — strictly more correct). The source is intentionally left unmigrated for the normal launch-time pass. Migration runs before the #209 snapshot write so the snapshot reflects migrated triggers.

### #234 (medium) — reminder minutes field could crash saveProfile
The minutes→seconds conversion goes through `BlockedProfileView.reminderTimeSeconds(enabled:minutes:)`, clamped to `0...UInt32.max/60`, plus a matching `.onChange` clamp on the TextField (mirroring the adjacent reminder-message clamp). Bounds were chosen deliberately after review: they preserve **every previously savable value** (e.g. a weekly 10 080-minute reminder survives unchanged) and only clamp input that would have trapped `UInt32(_:)` — negative values or ≥ 71 582 789 minutes.

### Preliminary commit — Xcode 26.6 build break (pre-existing)
`xcodebuild test` fails on unmodified `main` under Xcode 26.6: four Swift 6 `sending 'activity' risks causing data races` errors in `LiveActivityManager`. Commit 1 fixes this by looking the `Activity` up by ID inside nonisolated static helpers (the pattern `restoreExistingActivity` already uses) so the non-Sendable reference never leaves the main actor. Without it no tests can build or run on current toolchains.

## Testing

- New regression tests (TDD, RED observed before each fix): `ProfileSaveSnapshotTests` (3), `CloneProfileTests` (7, incl. pinning tests that the source profile and V2 clones are untouched), `ReminderTimeConversionTests` (7).
- Full `FoqosTests` suite green on iPhone 17 simulator: **429 passed, 0 failed**.
- `swift-format lint` clean on all touched files.

## Review

A 7-agent adversarial review (per-fix dimensions, concurrency, regression sweep, test quality; findings adversarially verified) ran before commit. Its one confirmed finding — the original 1440-minute clamp silently rewriting legitimate pre-existing reminder values — was fixed (see #234 above, bounds `0...UInt32.max/60`). Human review requested per AGENTS.md.

## Out of scope / follow-up

- `SyncModels.swift:198-199` performs the same unchecked `UInt32(reminderInt)` conversion on inbound CloudKit record data — same defect class as #234, worth its own issue.
- V1→V2 migration paths (`ProfileMigrationUtil`, post-session migration) still don't re-snapshot after mutating trigger fields — same class as #198, largely masked by the legacy-schedule fallback (low severity, flagged by review but not confirmed as user-visible).

## Summary by Sourcery

Strengthen the profile save and clone pipeline, fixing snapshot, migration, sync, and reminder-time issues and updating live activity handling for Swift 6 toolchains.

Bug Fixes:
- Ensure profile trigger/schedule changes refresh the shared snapshot so background blocking uses up-to-date data.
- Guarantee cloned profiles are migrated appropriately from legacy schemas and receive immediate snapshots and sync pushes.
- Prevent crashes when saving reminder times by clamping free-form minute input to a safe UInt32 range.

Enhancements:
- Adjust live activity update/end handling to satisfy Swift 6 sendability rules while keeping activity references confined to the main actor context.

Tests:
- Add regression test suites covering snapshot updates on trigger saves, cloning behavior across schema versions, and reminder time conversion edge cases.

### CODE REVIEW

**Comment by mnbf9rca** [comment]
Both follow-up candidates from the PR body are verified and now filed: #265 (unchecked `UInt32(reminderInt)` in `SyncModels.swift:199` on inbound CloudKit data — remote twin of #234, medium) and #266 (V1→V2 migration paths never re-snapshot — same class as #198, low). Both added to epic #263 (#265 folded into bundle A3). Nothing further blocking this PR.

File: Foqos/Models/TriggerConfigurationModel.swift [hunk 1]
```diff
@@ -109,5 +109,10 @@ final class TriggerConfigurationModel: ObservableObject {
     profile.stopQRCodeId = stopQRCodeId
     profile.startSchedule = startSchedule
     profile.stopSchedule = stopSchedule
+
+    // Refresh the app-group snapshot after trigger persistence — the snapshot
+    // written earlier by createProfile/updateProfile predates these fields, and
+    // the FoqosDeviceMonitor extension enforces schedules purely from it (#198)
+    BlockedProfiles.updateSnapshot(for: profile)
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply Foqos/Models/TriggerConfigurationModel.swift]: Fixed. Reworded — the comment now states the trigger fields are set on the in-memory model and that SwiftData persistence happens later in the caller (finalizeSave), instead of the misleading "after trigger persistence". (ff74fba)

**Comment by mnbf9rca** [comment]
Re: general feedback from @sourcery-ai

**"Move the clone sync push (or a hook for it) into the model layer"** — Won't fix in this PR. `BlockedProfiles` deliberately has zero `SyncCoordinator` references: pushing is a view/coordinator-layer concern throughout the codebase (`createProfile`/`updateProfile` don't push either — `finalizeSave` does, and the delete flow calls `deleteProfileFromSync` from the view). `cloneProfile` has exactly one caller, so a hypothetical future non-UI caller is speculative. Centralizing sync-on-mutation is a real architectural question, but it belongs with the sync bundles (A×/E2) on epic #263, not this defect-fix PR.

**"Extract a shared `activity(withId:)` helper for the duplicated `Activity.activities` lookup"** — Fixed. Extracted `private nonisolated static func activity(withId:)` used by both `updateActivity` and `endActivity`; the non-Sendable/lookup-by-ID doc comment now sits directly above it. (ff74fba)
